### PR TITLE
Automatic Autoupdate

### PR DESCRIPTION
**Pull request by MCOfficer**
- Scheduled workflow every 3h (up for discussion)
- For each manifest:
	- Run the autoupdate script
	- if there are changes, push them to a new branch named specific to the manifest + version
	- if the branch already exists, do nothing
	- open a PR for the branch and label it "Autoupdate"

Test runs:

https://github.com/MCOfficer/endless-sky-plugins-1/actions/runs/1139274655
https://github.com/MCOfficer/endless-sky-plugins-1/pulls

Before merging, create an Autoupdate PR label.

### CODE REVIEW

File: .github/workflows/autoupdate.yml
Comment (tehhowch) [inline .github/workflows/autoupdate.yml:44]: What about multiple releases without an update? Release - oops, fix bug - release again, all before the previous is merged?
Basically thinking like dependabot behavior.

File: .github/workflows/autoupdate.yml
Comment (tehhowch) [inline .github/workflows/autoupdate.yml:38]: I think a dedicated prefix would be good, as well as using a character disallowed in plugin names (`/`) to separate the version. That will support determining which existing branches target the same plugin via `cut`
```suggestion
            branchname=$(echo "auto/$name/$version" | sed 's/[^\w\d]/_/g') # "auto/Plugin_Name/1_5_3"
```

File: .github/workflows/autoupdate.yml
Comment (tehhowch) [inline .github/workflows/autoupdate.yml:53]: We should fill the PR body with a link to the version diff where possible, to make it dead-simple for a maintainer to inspect the changes before merging.

**Review by tehhowch — COMMENTED** [review]
A good step forward 👍

We should consider exporting the parse and integration test actions from the ES repo into their own action repos, so that we can use them here as plugin validation.

**Comment by MCOfficer** [comment]
Not sure what else to do about this. If you have any conceptual ideas, hit me before diving into the code, otherwise this is ready for review.

File: .github/workflows/autoupdate.yml
Comment (MCOfficer) [reply .github/workflows/autoupdate.yml:44]: The branches are version-specific, so it would push to a new branch and open a new PR. This condition is to prevent possible checkout failures in case the workflow runs again while the PR isn't merged yet. In this case it would update the file on master, then try to checkout the branch where this exact change is already committed.

File: .github/workflows/autoupdate.yml
Comment (MCOfficer) [reply .github/workflows/autoupdate.yml:38]: Very good suggestion, this is much more tidy.

File: .github/workflows/autoupdate.yml
Comment (MCOfficer) [reply .github/workflows/autoupdate.yml:45]: What would lead to a failure here? It will never check out existing branches, and i've made sure to sanitize the branch names.

File: .github/workflows/autoupdate.yml
Comment (tehhowch) [reply .github/workflows/autoupdate.yml:53]: Yeah, I'm suggesting the intermediate PR gets closed and its branch deleted.

No point having a PR for version 1.0.0 and 1.0.1 both open.

Of course, having said that, there is a case to be made for keeping both: if we implement plugin-game validation, it'd be possible for the earlier PR to pass the checks, but the second PR to fail the checks. In that case we wouldn't want to close the "intermediate" PR because the latest version introduces a (presumably different) bug.

Comment (MCOfficer) [reply .github/workflows/autoupdate.yml:53]: My first thought was "that's way too complicated to be worth the effort" but thinking about it... a simple `gh pr close -d "$old_branch"` should do it.

So the argument is your concerns about specific versions breaking, versus having to manually close PRs once in a blue moon.

(Regarding deleting branches, i plan to follow up with an action that gets rid of all `auto/` branches once their PR is closed)

File: .github/workflows/autoupdate.yml
Comment (MCOfficer) [reply .github/workflows/autoupdate.yml]: turns out this was correct, we want to (supposedly) start a subshell and check if there's anything on stdout. The exit code is 0 either way.

Guess that's what i get from stealing commands from stackoverflow.

File: .github/workflows/autoupdate.yml
Comment (MCOfficer) [reply .github/workflows/autoupdate.yml]: Here's the bare minimum: https://github.com/MCOfficer/endless-sky-plugins-1/pull/59

File: .github/workflows/autoupdate.yml
Comment (tehhowch) [reply .github/workflows/autoupdate.yml:32]: You could pipe to grep and test that:
```bash
if git status --porcelain | grep -q "."; then
  # dirty dir
else
  # clean dir
```
Note that grep doesn't like it if you pass _nothing_, so you wouldn't want to do `if git status --porcelain | grep -v "."; then` to detect the case where the directory is clean. You'd negate the command (`if ! git...`), rather than the grep match for that scenario.

Comment (MCOfficer) [reply .github/workflows/autoupdate.yml:32]: That would work, but is it better than this?

File: .github/workflows/autoupdate.yml
Comment (tehhowch) [inline .github/workflows/autoupdate.yml:28]: Should this be origin/master?

```suggestion
          git reset --hard origin/master
```

File: .github/workflows/autoupdate.yml
Comment (tehhowch) [inline .github/workflows/autoupdate.yml:55]: Probably don't need to set the upstream but w/e

File: .github/workflows/autoupdate.yml
Comment (tehhowch) [inline .github/workflows/autoupdate.yml:47]:
```suggestion
            if git branch -al | grep -Fq "remotes/origin/$branchname"; then
```

**Review by tehhowch — APPROVED** [review]
Nits.

File: .github/workflows/autoupdate.yml
Comment (MCOfficer) [reply .github/workflows/autoupdate.yml:28]: Same thing - the `cron` trigger always checks out the default branch, and there's no possibility something is commited to the local master.

File: .github/workflows/autoupdate.yml
Comment (MCOfficer) [reply .github/workflows/autoupdate.yml:55]: It would throw an error otherwise.

**Pull request merged**
